flow/connectors/external_metadata/store.go

```diff
@@ -118,29 +121,49 @@ func (p *PostgresMetadataStore) SetupMetadata() error {
 
 	// create the last sync state table
 	_, err = p.conn.Exec(p.ctx, `
-		CREATE TABLE IF NOT EXISTS `+p.QualifyTable(lastSyncStateTableName)+` (
+		CREATE TABLE IF NOT EXISTS `+p.QualifyTable(lastSyncStateTableName)+`(
 			job_name TEXT PRIMARY KEY NOT NULL,
 			last_offset BIGINT NOT NULL,
-			updated_at TIMESTAMP NOT NULL DEFAULT NOW(),
+			updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
 			sync_batch_id BIGINT NOT NULL,
 			normalize_batch_id BIGINT
-		)
-	`)
+		)`)
 	if err != nil && !utils.IsUniqueError(err) {
 		p.logger.Error("failed to create last sync state table", slog.Any("error", err))
 		return err
 	}
 
+	_, err = p.conn.Exec(p.ctx, `
+		CREATE TABLE IF NOT EXISTS `+p.QualifyTable(qrepTableName)+`(
+			job_name TEXT NOT NULL,
+			partition_id TEXT NOT NULL,
+			sync_partition JSON NOT NULL,
+			sync_start_time TIMESTAMPTZ NOT NULL,
+			sync_finish_time TIMESTAMPTZ NOT NULL DEFAULT NOW()
+		)`)
+	if err != nil && !utils.IsUniqueError(err) {
+		p.logger.Error("failed to create qrep metadata table", slog.Any("error", err))
+		return err
+	}
+
+	_, err = p.conn.Exec(p.ctx,
+		`CREATE INDEX IF NOT EXISTS ix_qrep_metadata_partition_id ON `+
+			p.QualifyTable(qrepTableName)+
+			` USING hash (partition_id)`)
+	if err != nil && !utils.IsUniqueError(err) {
+		p.logger.Error("failed to create qrep metadata index", slog.Any("error", err))
+		return err
+	}
+
```

Comment:
Eventhub and S3 don't need this I think. Can we have a new `SetupDWHMetadata()` method which sets these tables up or just pass a flag

Reply:
Would rather have empty tables than differing schemas, we're considering moving metadata table setup into migration files

Reply:
Got it. Makes sense